gensim/models/ensemblelda.py

```diff
@@ -0,0 +1,1268 @@
+            If True, the models in the ensemble are
+            deleted after training and only the
+            total topic word distribution is kept
+            to save memory.
+
+            Defaults to True. When False, trained
+            models are stored in a list in self.tms,
+            and no models that are not of a gensim
+            model type can be added to this ensemble
+            using the add_model function.
+
+            If False, any topic term matrix can be
+            supllied to add_model.
+        min_samples : number, optional
+            Required number of nearby topics for
+            a topic to be considered as 'core' in
+            the CBDBSCAN clustering.
+        masking_method : {'mass', 'rank}, optional
+            One of "mass" (default) or "rank" (faster).
+        masking_threshold : float, optional
+            Default: None, which uses 0.11 for
+            masking_method "rank", and 0.95
+            for "mass".
+        distance_workers : number, optional
+            When distance_workers is None, it defaults to
+            os.cpu_count() for maximum performance.
+            Default is 1, which is not multiprocessed.
+            Set to > 1 to enable multiprocessing.
+        **gensim_kw_args
+            All the parameters for the gensim model,
+            that is used for each model in the ensemble,
+            can be provided as additional keyword arguments.
+            https://radimrehurek.com/gensim/models/ldamodel.html
+
+        """
+
+        if "id2word" not in gensim_kw_args:
+            gensim_kw_args["id2word"] = None
+        if "corpus" not in gensim_kw_args:
+            gensim_kw_args["corpus"] = None
+
+        # dictionary. modified version of from gensim/models/ldamodel.py
+        # error messages are copied from the original gensim module [1]:
+        # will create a fake dict if no dict is provided.
+        if gensim_kw_args["id2word"] is None and not gensim_kw_args["corpus"] is None:
+            logger.warning("no word id mapping provided; initializing from corpus, assuming identity")
+            gensim_kw_args["id2word"] = utils.dict_from_corpus(gensim_kw_args["corpus"])
+        if gensim_kw_args["id2word"] is None and gensim_kw_args["corpus"] is None:
+            raise ValueError("at least one of corpus/id2word must be specified, to establish "
+                             "input space dimensionality. Corpus should be provided using the "
+                             "keyword argument corpus.")
+
+        if type(topic_model_kind) == str:
+            self.topic_model_kind = {
+                "lda": ldamodel.LdaModel,
+                "ldamulticore": ldamulticore.LdaMulticore
+            }[topic_model_kind]
+        else:
+            self.topic_model_kind = topic_model_kind
+
+        # Store some of the parameters
+        self.num_models = num_models
+        self.gensim_kw_args = gensim_kw_args
+
+        # some settings affecting performance
```

Comment:
As above. These comments may be more helpful in the docstring for the parameters.

Reply:
done